src/arkreactor/Compiler/Parser.cpp

```diff
@@ -126,6 +128,14 @@ namespace Ark
             if (token.token == ")")
                 return block;
 
+            // check for unexpected keywords between expressions
+            if ((token.type == TokenType::Operator ||
+                 token.type == TokenType::Identifier ||
+                 token.type == TokenType::Number ||
+                 token.type == TokenType::String) &&
+                tokens.front().type == TokenType::Keyword)
+                throwParseError("Unexpected keyword `" + tokens.front().token + "' in the middle of an expression", token);
```

Comment:
The last argument here should be `tokens.front()` instead of `token`, so that the error context is related to the unexpected keyword and not what comes before it.